src/storage/arrow_serializer.cpp

```diff
@@ -43,9 +43,8 @@ void ArrowSerializer::AssembleMetadataBuffer(std::ofstream &outfile, flatbuf::Me
 void ArrowSerializer::WriteSchemaMessage(std::ofstream &outfile, std::unordered_map<col_id_t, int64_t> *dictionary_ids,
                                          std::vector<type::TypeId> *col_types,
                                          flatbuffers::FlatBufferBuilder *flatbuf_builder) {
-  RawBlock *block = data_table_.blocks_.front();
   const BlockLayout &layout = data_table_.accessor_.GetBlockLayout();
-  ArrowBlockMetadata &metadata = data_table_.accessor_.GetArrowBlockMetadata(block);
+  ArrowBlockMetadata &metadata = data_table_.accessor_.GetArrowBlockMetadata(*(data_table_.GetBlocks().begin()));
```

Comment:
nit:  It's fine the way you have it now, but I would generally prefer the original form because it is easier to read and digest the different steps when reading it and is also more friendly for debugger sessions.

Reply:
fixed

Reply:
done